profiler/profiler_test.go

```diff
@@ -184,58 +184,43 @@ func TestStartStopIdempotency(t *testing.T) {
 // TestStopLatency tries to make sure that calling Stop() doesn't hang, i.e.
 // that ongoing profiling or upload operations are immediately canceled.
 func TestStopLatency(t *testing.T) {
-	t.Skip("broken test, see issue #1294")
-
-	t.Run("stop-profiles", func(t *testing.T) {
-		Start(
-			WithPeriod(time.Hour),
-			CPUDuration(time.Hour),
-		)
-
-		// give profiling time to start
-		time.Sleep(50 * time.Millisecond)
-		start := time.Now()
-		Stop()
-		elapsed := time.Since(start)
-		// CPU profiling polls in 100 millisecond intervals and this can't be
-		// interrupted by pprof.StopCPUProfile, so we can't guarantee profiling
-		// will stop faster than 200 milliseconds (one interval to read
-		// remaining samples, second interval to detect profiling has stopped).
-		// Set a conservative upper bound on that stop time plus profile
-		// serialization.
-		if elapsed > 500*time.Millisecond {
-			t.Errorf("profiler took %v to stop", elapsed)
+	received := make(chan struct{})
```

Comment:
NIT: Maybe give this channel a capacity of 1 to avoid the very unlikely race of `<-received` on line 206 getting stuck because the http handler fired much earlier than expected. (Again: Extremely unlikely)

Reply:
I think it's safe to have the channel unbuffered. The handler should be called repeatedly, once per profiling period. If the handler somehow fired before the main goroutine got to line 206, it would just fire again the next profiling period.